src/lib/elastic.ts

```diff
@@ -61,37 +65,39 @@ function adjustQuery (esQuery, entityType, config) {
   return esQuery
 }
 
-function getResponseObject (result) {
+export function getResponseObject (result) {
   if (result.body) { // differences between ES5 andd ES7
     return result.body
   } else {
     return result
   }
 }
 
-function getHits (result) {
+export function getHits (result) {
   if (result.body) { // differences between ES5 andd ES7
     return result.body.hits.hits
   } else {
     return result.hits.hits
   }
 }
 
-function getClient (config) {
-  const esConfig = { // as we're runing tax calculation and other data, we need a ES indexer
-    node: `${config.elasticsearch.protocol}://${config.elasticsearch.host}:${config.elasticsearch.port}`,
-    apiVersion: config.elasticsearch.apiVersion,
+export function getClient (config: IConfig): Client {
+  const esConfig: ClientOptions = { // as we're runing tax calculation and other data, we need a ES indexer
+    node: `${config.get<string>('elasticsearch.protocol')}://${config.get<string>('elasticsearch.host')}:${config.get<number>('elasticsearch.port')}`,
     requestTimeout: 5000
   }
-  if (config.elasticsearch.user) {
-    esConfig.auth = config.elasticsearch.user + ':' + config.elasticsearch.password
+  if (config.get<string>('elasticsearch.user')) {
+    esConfig.auth = {
+      username: config.get<string>('elasticsearch.user'),
+      password: config.get<string>('elasticsearch.password')
+    }
   }
   return new es.Client(esConfig)
 }
 
-function putAlias (db, originalName, aliasName, next) {
+export function putAlias (db: Client, originalName: string, aliasName: string, next) {
```

Comment:
Here I need your help.

What type should next be?

Reply:
it's a callable - function kind of :)

Reply:
I thing I have found for all callbacks the right type for it.